robot_calibration/src/sensor_data_generator.cpp

```diff
@@ -1,43 +1,85 @@
 #include "ros/ros.h"
+#include "ros/package.h"
 #include "sensor_msgs/JointState.h"
 #include <cmath>
 #include "yaml-cpp/yaml.h"
 #include <fstream>
+#include "boost/filesystem.hpp"
 
-
-void calculateAndWriteData(const sensor_msgs::JointState::ConstPtr& msg)
+class Listener
 {
-  double angles[2];
-  angles[0] = msg->position[0];
-  angles[1] = msg->position[1];
+  public:
+    void calculateAndWriteData(const sensor_msgs::JointState::ConstPtr& msg);
+    std::string filename;
+    double theta_1, theta_2, theta_1_offset, theta_2_offset;
+    double ee_position[2];
+    Listener(double t_1_offset, double t_2_offset)
+    {
+      filename = ros::package::getPath("robot_calibration");
+      if(filename != "")
+      {
+        filename += "/sensor_data";
+        boost::filesystem::create_directory(filename);
+        filename += "/sensor_data_" + std::to_string((int)ros::Time::now().toSec()) + ".yaml";
+      }
+      theta_1 = -1.0;
+      theta_2 = -1.0;
+
+      ee_position[0] = 1000000000;
+      ee_position[1] = 1000000000;
```

Comment:
This is an issue i was supposed to talk to you about before I submitted, sorry. Im not 100% sure what to initialise the class member variables to. In the callback function, it does a check to make sure the angles received in the joint states message are different to those stored in the member variables before it writes. So I essentially just want to initialise it to something impossible like 7 radians. (Dont ask me why they're -1, i dont even know). But i wanted to check with you to see if that's acceptable or if there's another way to do it that's considered better practice.

Reply:
The main point of my initial comment was to make sure you explain why you're doing things. You can just assign these to the [max float value](http://www.cplusplus.com/reference/limits/numeric_limits/) and add a comment saying you're initialising these to an unreasonable value, so that they'll be changed once a joint state message is received.